src/infra/ssl/sslQueryRepo.go

```diff
@@ -136,73 +87,155 @@ func (repo SslQueryRepo) SslPairFactory(
 
 	return entity.NewSslPair(
 		hashId,
-		sslHostname,
+		sslVhosts,
 		certificate,
 		sslPrivateKey,
 		chainCertificates,
 	), nil
 }
 
+func (repo SslQueryRepo) getHostnameFromCertFilePath(
+	certFilePath string,
+) (valueObject.Fqdn, error) {
+	certFilePathWithoutBase := filepath.Base(certFilePath)
+	certFileNameWithoutExt := strings.TrimSuffix(certFilePathWithoutBase, ".crt")
+
+	return valueObject.NewFqdn(certFileNameWithoutExt)
+}
+
+func (repo SslQueryRepo) getCertFilesPathWithSymlinkVhosts(
```

Comment:
Cara, eu só sei o que isso aqui faz pois foi eu te passei o algo. No entanto, sua execução deixou a desejar demais, eu simplesmente não consigo fazer sentido algum do que tá rolando aqui.

Reply:
- O isDirPath foi uma validação necessária enquanto a [pr 19](https://github.com/speedianet/os/pull/19) não é mergeada, ou seja, o primeiro item de um FilesQueryRepo.Get(), quando o caminho for um diretório, sempre será o diretório em si. Nesse caso, o primeiro item sempre será o "/app/conf/pki". Isso foi ajustado no [pr 19](https://github.com/speedianet/os/pull/19) a pedido do Roger;
- O isCertFile é para verificar se o arquivo que está vindo na iteração atual é um arquivo de certificado. Lembro de você dizendo que eu filtraria por eles (arquivos com extensão "crt");
- O isSymlink era uma validação que eu não queria ter feito, mas fui obrigado a fazer por conta do os.Readlink().
    - O os.Readlink() é o responsável por buscar o caminho do arquivo target de um symlink. Contudo, se o caminho informado nessa função não for um symlink, ele dá erro. Ou seja, antes de passar o caminho pra lá, eu preciso validar se ele é um symlink ou não.
    - Na parte do próprio isSymlink eu não consigo garantir que o meu arquivo target (não é symlink) sempre vai vir antes dos arquivos que são symlinks, ou seja, eu preciso verificar se dentro do meu map eu já tenho aquele target mapeado como chave antes de setar como um slice vazio. Se eu já setar ele como um slice vazio sem fazer essa checagem, eu posso dar overwrite em um slice que já havia sido populado anteriormente.
        - Ex.: tenho os arquivos "matheus.net.crt" e o "speedia.net.crt". O "matheus.net.crt" é symlink do "speedia.net.crt", mas o "speedia.net.crt" na listagem do FileQueryRepo.Get() vem depois do "matheus.net.crt". Se isso acontecer e eu não tiver a validação de se o target já foi setado antes no meu map, eu vou acabar setando um slice vazio e perder o slice de symlinks anterior com o "matheus.net.crt" lá dentro.
- Eu repito a validação lá em baixo também, porque ali significa que o cara não é um symlink, portanto eu ainda assim preciso validar se existe uma chave no meu map com o target e, se não tiver, eu adiciono o target ao map com uma slice vazia. Depois populo com os meus symlinks.

Reply:
Revi meu código e mudei um pouco como ele tá funcionando nesse método.

- Antes da verificação do isSymlink, eu crio uma variável chamada targetCertFilePathStr que recebe, de cara, o valor do certFilePathStr;
- No isSymlink, agora eu estou validando se ele é de fato um. Caso for, eu busco o target desse symlink e atribuo ao targetCertFilePathStr;
- Faço a validação de se o target existe dentro do meu map. Caso não exista, eu adiciono ele junto de um slice vazio;
- Verifico se certFilePathStr == targetCertFilePathStr.
    - Caso for, significa que se trata do arquivo target, portanto eu não vou adiante.
    - Caso não for, significa que se trata de um symlink, portanto eu extraio o vhost como Fqdn do meu file path e adiciono no slice de symlinks do meu map.